Mathlib/MeasureTheory/Measure/GiryMonad.lean

```diff
@@ -174,11 +179,22 @@ theorem bind_dirac {f : α → Measure β} (hf : Measurable f) (a : α) : bind (
   erw [bind_apply hs hf, lintegral_dirac' a ((measurable_coe hs).comp hf)]
   rfl
 
+@[simp]
 theorem dirac_bind {m : Measure α} : bind m dirac = m := by
   ext1 s hs
   simp only [bind_apply hs measurable_dirac, dirac_apply' _ hs, lintegral_indicator 1 hs,
     Pi.one_apply, lintegral_one, restrict_apply, MeasurableSet.univ, univ_inter]
 
+@[simp]
+lemma dirac_bind_eq_map (m : Measure α) {f : α → β} (hf : Measurable f) :
```

Comment:
Can you explain why the other lemma you introduce is `bind_const`, while this one is `dirac_bind_eq_map` instead of `bind_dirac_eq_map`? (I notice that the previous lemma is also `dirac_bind` and not `bind_dirac`). That's an honest question, I'm not really familiar with the monadic language.

Reply:
I don't know the logic either. When I first wrote the lemma, I named it `bind_dirac_eq_map`, but then I saw `bind_dirac` and `dirac_bind` and changed the name to match the closest one. But I agree that it does not match `bind_const` anymore (nor `bind_zero_left/right` etc.). Actually `dirac_bind` looks odd compared to the other names in that file.

Reply:
Then maybe it's worth renaming `dirac_bind` to `bind_dirac` for coherence.

Reply:
I exchanged the names. That's not something for which I can use deprecations to indicate the change, I think. Let's hope they were not used much.